src/components/SubTask/components/SubTaskList/components/subTaskItem/index.js

```diff
@@ -0,0 +1,193 @@
+import { Component } from "react";
+import * as React from "react";
+import { connect } from "react-redux";
+import { OverlayTrigger, Popover, Button } from "react-bootstrap";
+import { THREEDOTS } from "assets/Icon";
+import { updateTaskItem } from "../../../../../TaskList/components/TaskItem/actions";
+import "../../scroll.css";
+
+class SubTaskItem extends Component {
+  constructor(props) {
+    super(props);
+    const { subTask } = this.props;
+    this.state = {
+      isEditMode: false,
+      title: subTask.description,
+    };
+    this.titleInput = React.createRef();
+  }
+
+  getSubTaskIndex = (subTasks, description) => {
+    for (let i = 0; i < subTasks.length; ) {
+      if (subTasks[i].description === description) {
```

Comment:
What happens if there are two subtasks with identical descriptions or is this something we don't need to worry about?

Reply:
yeah, as Aamir mentioned we should probably add an IdObject to subtasks to uniquely identify them
also, i've discovered a findIndex function for array that will be useful for these cases